Fixed - Project specific reset password

ravindra-tc opened the pull request:
### Description

This pull request addresses the project-specific reset password issue in the Medplum project. Specifically, the fix involves adding the project to the user filters to resolve the reset password issue. 

### Test Cases

I have run test cases for the resetPassword functionality. However, I've identified an issue with the `Reset Password › Custom reCAPTCHA site key success` test case. Upon investigation, it appears that the test is failing because the `projectId` is provided in the request body, but the associated project is missing in the user resource. The root cause is that the project is not assigned to the user resource at the time of creating the user. 
[Test - Reset Password › Custom reCAPTCHA site key success](https://github.com/medplum/medplum/blob/01fdd05735ad1794884c349a876e68df01fe44b2/packages/server/src/auth/resetpassword.test.ts#L199-L245).

### Action Items

- [x] Fix the reset password issue by adding the project to the user filters.
- [ ] Review by @rahul1 and @codyebberson.

Please look at it and take it forward.

@rahul1 @codyebberson

#3340

rahul1 commented on packages/server/src/auth/resetpassword.ts:
```diff
@@ -57,16 +57,29 @@ export async function resetPasswordHandler(req: Request, res: Response): Promise
+    operator: Operator.EQUALS,
+    value: 'Project/' + req.body.projectId, // Set the project value for equality matching
   });
+}
```
For safety, add an `else` clause here that filters for `project` is missing: 
```
    code: 'project',
    operator: Operator.MISSING,
    value: true
  }
```

This will enforce that a missing project parameter only resets server-level useres

rahul1 reviewed (changes requested):
Thanks so much for the contribution @ravindra-tc . 

This looks good, but I made one request for an `else` clause. 
It also looks like there is a broken test. 

Running `npm run prettier` as and fixing those issues will make sure the build passes

ravindra-tc commented:
@rahul1 I've incorporated an `else` clause for the project-based filter. Furthermore, I've resolved the issues in the broken test case and introduced two new test cases to cover the new scenario.
